.github/CONTRIBUTING.md

```diff
@@ -1,36 +1,84 @@
 # Contributing
 
+## Workflow
+
+1. Fork, clone, and select the **main** branch.
+2. Create a new branch in your fork.
+3. Make your changes.
+4. Commit your changes, and push them.
+5. Submit a Pull Request [here]!
+
+## Contributing to the guides
+
+The guides for all Sapphire Project use [mdx] formatting. At its core this is the exact same as regular markdown, and it supports all your regular GitHub flavoured markdown syntax. Where it differs from this regular markdown however, is that it _also_ supports inline JSX tags. These are used for advanced and optimal layout on `<docs site address pending>`. If you want to learn how and what kind of JSX you can embed into your [mdx] then please consult the [mdx documentation][]
+
+When writing guides there are some rules to follow:
+
+1. All files should have the `.mdx` file extension.
+2. Folder names are allowed to have spaces.
+3. Each word in a folder name should started with a Capital letter.
+4. File names should _never_ have spaces.
+5. File names that consist of multiple words should be PascalCased.
+6. Information in a guide page should be generally useful to the majority of people.
+    - The single exception to this goes to files in the "Advanced" folder, which can cover any type of advanced topics.
+7. There are several variables that will be replaced when parsing the `.mdx` file for the documentation website. Variables are denoted by the pattern `{@variableName (parameter)}`. The supported variables are:
+    - `{@branch}` - for the current branch name.
+    - `{@link <TSDoc symbol name>}` - wherein `TSDoc symbol name` is a reference to some TypeScript interface / class / function etc parsed from TSDoc. This will link the API page of that symbol.
```

Comment:
```suggestion
    - `{@link <TSDoc symbol name>}` - wherein `<TSDoc symbol name>` is a reference to some TypeScript interface / class / function etc parsed from TSDoc. This will link the API page of that symbol.
```